src/stransport_stream.c

```diff
@@ -150,9 +156,9 @@ static OSStatus read_cb(SSLConnectionRef conn, void *data, size_t *len)
 	requested = left = *len;
 	do {
 		ret = git_stream_read(io, data + (requested - left), left);
-		if (ret < 0) {
+		if (ret <= 0) { /* There was an error or nothing could be read because stream is closed */
 			*len = 0;
-			return -1;
+			return errSSLClosedAbort;
```

Comment:
The above Apple link is actually from an older Security project version and is not around in the version shipping on 10.10. However, there's another use of the API at http://www.opensource.apple.com/source/Security/Security-55179.13/libsecurity_ssl/sslViewer/ioSock.c which has socket wrapper implementations of `SocketRead()` and `SocketWrite()` to be used with SSLSetIOFuncs().